### ISSUE TITLE
Bug: Transaction stuck on pending

### ISSUE THREAD

**Issue by alfetopito** [gnosis/safe-cmm-app-react#239]
# Description

A user reports there's a pending tx stuck on his Safe.
User is also not able to deploy new strategies. Possibly related.

# To Reproduce:

Not able to reproduce

# Reported user steps

1. User deployed a strategy
2. Transaction was stuck on metamask, so the user cancelled it
3. Transaction still shows as pending on his Safe, even after reloading, different browser and so on
4. User reports they did not succeed to deploy new strategies since then

# Expected Behaviour

There are no pending txs after cancelling them and new txs can be deployed without issues

# Actual Behaviour

Even after cancelling the Metamask transaction, CMM pending tx is still present.
We can see that by checking the response of https://safe-transaction.xdai.gnosis.io/api/v1/safes/0x45323abC6e9FCf87654001B893EA4f41bb4a8190/transactions/?nonce__gte=6&executed=false

Also, user is not able to deploy a new strategy.

When we look at the console errors though, there's an error when loading the txs (see screen shot on next section).

Still, the tx that fails is one that has been executed, not the pending one:

Failed tx: [0xe4ae065861f9a501781f14a1ed0949290328cdea989ec29081e421fe916ce2a0](https://dashboard.tenderly.co/tx/xdai/0xe4ae065861f9a501781f14a1ed0949290328cdea989ec29081e421fe916ce2a0)

User's Safe has no pending CMM tx to be signed.
The Safe has a single owner, thus it shouldn't have any pending tx anyway.

# Screenshots/Videos

## Pending tx
![](https://cdn.discordapp.com/attachments/639851872487735316/789523313835376690/Screen_Shot_2020-12-18_at_8.01.21_AM.png)

## Safe txs page
![](https://cdn.discordapp.com/attachments/639851872487735316/789523969140195368/Screen_Shot_2020-12-18_at_8.05.59_AM.png)

## Deploy page error
![](https://cdn.discordapp.com/attachments/639851872487735316/789530911799771166/Screen_Shot_2020-12-18_at_8.33.34_AM.png)

## Error logs
![](https://cdn.discordapp.com/attachments/639851872487735316/789535851284922408/Screen_Shot_2020-12-18_at_8.53.13_AM.png)

![](https://cdn.discordapp.com/attachments/639851872487735316/789536729874038784/Screen_Shot_2020-12-18_at_8.56.21_AM.png)

# Environment
**Desktop:**
 - Browser Chrome
 - Version 0.4.6

# Background

N/A

**Comment by alfetopito** [comment]
A few points to note raised by Richard:

- The pending tx wasn't confirmed, so to clear it out, the user has to perform any action that updates the Safe nonce.
- The way we are fetching pending txs is using an internal Safe endpoint that will change in the near future
- When it changes though, the txs will be part of the safe SDK, so we won't be left without alternatives
- The pending tx has `safeTxGas` value set to `0`, which indicates that the tx would fail

**Comment by alfetopito** [comment]
Also:

- The console error log in question is happening on the oldest deploy strategy transaction (14/12/2020 07:01:10)
- The error message `Unexpected token < in JSON...` seems like something that was expected to be JSON returned an HTML instead. 
- We don't fetch any JSON directly, so it might be a web3 call?
- Also, HTML response when JSON is expected generally means the response is an error, and not a nice one
- This might indicate that the error is not with the pending strategy at all, but instead with something in the setup
- It could be an issue with the wallet provider's connection, maybe Infura network problems
- It could also be an extension conflicting/blocking some calls

**Comment by alfetopito** [comment]
Reached out to the user asking for more input but haven't heard back yet.
Will keep this on hold for now.
